src/txn/txn_rollback_to_stable.c

```diff
@@ -358,6 +360,84 @@ __txn_abort_newer_col_fix(
         __txn_abort_newer_insert(session, ins, rollback_timestamp);
 }
 
+/*
+ * __txn_abort_row_hs_updates_using_build_in_mem_page --
+ *     Abort updates on a history store using the in-memory build reconciled page of data store.
+ */
+static int
+__txn_abort_row_hs_updates_using_build_in_mem_page(WT_SESSION_IMPL *session, const void *image,
+  const uint8_t *addr, size_t addr_size, wt_timestamp_t rollback_timestamp)
+{
+    WT_DECL_RET;
+    WT_ITEM tmp;
+    WT_PAGE *mod_page;
+    WT_ROW *rip;
+    uint32_t i, page_flags;
+    const void *image_local;
+
+    /*
+     * Don't pass an allocated buffer to the underlying block read function, force allocation of new
+     * memory of the appropriate size.
+     */
+    WT_CLEAR(tmp);
+
+    image_local = image;
+
+    if (image_local == NULL) {
+        WT_RET(__wt_bt_read(session, &tmp, addr, addr_size));
+        image_local = tmp.data;
+    }
+
+    page_flags = WT_DATA_IN_ITEM(&tmp) ? WT_PAGE_DISK_ALLOC : WT_PAGE_DISK_MAPPED;
+    WT_ERR(__wt_page_inmem(session, NULL, image_local, page_flags, &mod_page));
+    tmp.mem = NULL;
+    WT_ROW_FOREACH (mod_page, rip, i)
+        WT_ERR_NOTFOUND_OK(
+          __txn_abort_row_with_hs_value(session, mod_page, rip, rollback_timestamp, false));
+    __wt_page_out(session, &mod_page);
+
+    if (0) {
+err:
+        if (mod_page != NULL)
+            __wt_page_out(session, &mod_page);
+
+        __wt_buf_free(session, &tmp);
+    }
+
+    return (ret);
+}
+
+/*
+ * __txn_abort_row_hs_updates_using_reconciled_pages --
+ *     Abort updates on a history store using the reconciled pages of data store.
+ */
+static int
+__txn_abort_row_hs_updates_using_reconciled_pages(
+  WT_SESSION_IMPL *session, WT_PAGE *page, wt_timestamp_t rollback_timestamp)
+{
+    WT_MULTI *multi;
+    WT_PAGE_MODIFY *mod;
+    uint32_t mul_entry;
+
+    if (page->modify == NULL)
+        return (0);
+
+    mod = page->modify;
+
+    if (mod && mod->rec_result == WT_PM_REC_REPLACE)
```

Comment:
You don't need to check for `mod` being valid - it's already handled above. If you do check, it should be a pointer equality: `mod != NULL`, rather than a bool like comparison.

Reply:
Fixed and remove the check for `mod`.